### ISSUE TITLE
Record requires ASM8

### ISSUE THREAD

**Issue by FilippoVigani** [getsentry/sentry-android-gradle-plugin#256]
_Android Gradle Plugin:_
**7.0.4**

_Gradle:_
**7.0.2**

_Optimizer:_
- [ ]  ProGuard
- [x]  R8

_Have you created the sentry.properties file?_
- [x]  Yes -> root folder of project
- [ ]  No

_SDKs:_
- [x]  sentry-android -> 5.5.2

---
I have the following issue:
I'm finding some incompatibilities when using this plugin with the latest version of moshi. See issue https://github.com/square/moshi/issues/1424#issuecomment-986821726 for details. I get the same error posted in that issue, i.e. `Record requires ASM8`. I can't tell if that's related, however when removing the dependency to the sentry gradle plugin I can compile without any issues.

**Steps to reproduce:**
- Add the following dependencies: 
```
implementation(platform("io.sentry:sentry-bom:5.5.2"))
implementation("io.sentry:sentry-android")
implementation("io.sentry:sentry-android-okhttp")
implementation("io.sentry:sentry-android-timber")
```

- Add plugin `io.sentry.android.gradle` version `3.0.0-beta-3`:
```
id("io.sentry.android.gradle") version "3.0.0-beta.3"
```

- Add configuration
```
sentry {
	includeProguardMapping.set(true)
	includeNativeSources.set(true)
	uploadNativeSymbols.set(true)
	autoUploadProguardMapping.set(true)
	autoUploadNativeSymbols.set(true)
	tracingInstrumentation {
		enabled.set(true)
	}
}
```
- Add moshi codegen and generate some adapters
```
implementation("com.squareup.moshi:moshi-kotlin:1.13.0")
kapt("com.squareup.moshi:moshi-kotlin-codegen:1.13.0")
implementation("com.squareup.retrofit2:converter-moshi:2.9.0")
```

- Run gradle sync and build project

**Actual result:**
- Build error:
```

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeExtDexProductionDebug'.
> Could not resolve all files for configuration ':app:productionDebugRuntimeClasspath'.
   > Failed to transform moshi-1.13.0.jar (com.squareup.moshi:moshi:1.13.0) to match attributes {artifactType=android-dex, asm-transformed-variant=productionDebug, dexing-enable-desugaring=true, dexing-is-debuggable=true, dexing-min-sdk=31, org.gradle.category=library, org.gradle.dependency.bundling=external, org.gradle.jvm.environment=standard-jvm, org.gradle.jvm.version=8, org.gradle.libraryelements=jar, org.gradle.status=release, org.gradle.usage=java-runtime, org.jetbrains.kotlin.platform.type=jvm}.
      > Execution failed for AsmClassesTransform: /Users/filippovigani/.gradle/caches/modules-2/files-2.1/com.squareup.moshi/moshi/1.13.0/da685586facab9eb5c4fb630ce248be14e7da21b/moshi-1.13.0.jar.
         > Record requires ASM8

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

**Expected result:**
- Build completes successfully

**Comment by romtsn** [comment]
thanks a lot for trying the beta version out, this is really valuable!

First question: are you using our [performance monitoring](https://docs.sentry.io/platforms/android/performance/)? If not, then you can simply set 
```kotlin
sentry {
 ...
 tracingInstrumentation {
   enabled.set(false) 
 }
}
```
and the issue should be fixed.

If you *are* using performance, then the only workaround for now is to force the ASM version to 8 as mentioned [here](https://issuetracker.google.com/issues/159151549#comment3) or use AGP 7.2.0-alpha06 if that's an option for you.

---
This has been fixed in AGP 7.2.0-alpha07, and according to [this comment](https://github.com/firebase/firebase-android-sdk/issues/3155#issuecomment-978103338) will be available as stable mid-February.

What we could do in the meantime (and for backward compatibility):
1. Force ASM version to 8 in our plugin (not sure if everything will work)
2. Exclude META-INF/ classes (looks like they are the [cause](https://github.com/firebase/firebase-android-sdk/issues/3155#issuecomment-1010341902))
3. Provide an ignorelist to ignore certain packages from instrumentation (note, that AGP 7.2.0 will have this option in the [standard API](https://developer.android.com/reference/tools/gradle-api/7.2/com/android/build/api/variant/Instrumentation#excludes()))

@marandaneto @bruno-garcia thoughts?

**Comment by marandaneto** [comment]
@romtsn Can we detect if AGP >= 7.2.0-alpha06 and if not, disable the integration that conflicts with it? also adding to the docs as min. required.?

**Comment by romtsn** [comment]
I think I'd rather go for excluding `META-INF/` files from instrumentation (I think it's possible to do on our side, but need to verify), because these classes are anyway not used at runtime.

**Comment by FilippoVigani** [comment]
I can confirm that disabling auto instrumentation fixes the issue. I was going to try out Room instrumentation but I can wait until AGP 7.2.0 gets released.
Thank you

**Comment by romtsn** [comment]
@FilippoVigani this is fixed in `3.0.0-beta.4` would you mind giving it another try? We're really looking forward to your feedback about Room instrumentation
